tests/ProxyManagerTest/Functional/RemoteObjectFunctionalTest.php

```diff
@@ -72,69 +81,84 @@ protected function getJsonRpcAdapter($expectedValue, string $method, array $para
-    {
-        $proxy = (new RemoteObjectFactory($this->getXmlRpcAdapter($expectedValue, $method, $params)))
+    public function testXmlRpcMethodCalls(
+        $instanceOrClassName,
+        string $method,
+        array $passedParams,
+        array $parametersForProxy,
+        $expectedValue
+    ): void {
+        $proxy = (new RemoteObjectFactory($this->getXmlRpcAdapter($expectedValue, $method, $parametersForProxy)))
             ->createProxy($instanceOrClassName);
 
         $callback = [$proxy, $method];
 
         self::assertIsCallable($callback);
-        self::assertSame($expectedValue, $callback(...$params));
+        self::assertSame($expectedValue, $callback(...$passedParams));
     }
 
     /**
      * @param string|object $instanceOrClassName
-     * @param mixed[]       $params
-     * @param mixed         $expectedValue
+     * @param string $method
+     * @param array $passedParams
+     * @param mixed[] $parametersForProxy
+     * @param mixed $expectedValue
      *
      * @dataProvider getProxyMethods
      *
      * @psalm-template OriginalClass of object
      * @psalm-param class-string<OriginalClass>|OriginalClass $instanceOrClassName
      */
-    public function testJsonRpcMethodCalls($instanceOrClassName, string $method, array $params, $expectedValue) : void
-    {
-        $proxy = (new RemoteObjectFactory($this->getJsonRpcAdapter($expectedValue, $method, $params)))
+    public function testJsonRpcMethodCalls(
+        $instanceOrClassName,
+        string $method,
+        array $passedParams,
+        array $parametersForProxy,
+        $expectedValue
+    ): void {
+        $proxy = (new RemoteObjectFactory($this->getJsonRpcAdapter($expectedValue, $method, $parametersForProxy)))
             ->createProxy($instanceOrClassName);
 
         $callback = [$proxy, $method];
 
         self::assertIsCallable($callback);
-        self::assertSame($expectedValue, $callback(...$params));
+        self::assertSame($expectedValue, $callback(...$passedParams));
     }
 
     /**
      * @param string|object $instanceOrClassName
-     * @param mixed         $propertyValue
+     * @param string $publicProperty
+     * @param mixed $propertyValue
      *
      * @dataProvider getPropertyAccessProxies
      *
      * @psalm-template OriginalClass of object
      * @psalm-param class-string<OriginalClass>|OriginalClass $instanceOrClassName
      */
-    public function testJsonRpcPropertyReadAccess($instanceOrClassName, string $publicProperty, $propertyValue) : void
+    public function testJsonRpcPropertyReadAccess($instanceOrClassName, string $publicProperty, $propertyValue): void
     {
         $proxy = (new RemoteObjectFactory($this->getJsonRpcAdapter($propertyValue, '__get', [$publicProperty])))
             ->createProxy($instanceOrClassName);
 
-        self::assertSame($propertyValue, $proxy->$publicProperty);
+        self::assertSame($propertyValue, $proxy->{$publicProperty});
```

Comment:
Why the `{``}`?

Reply:
I got a strange warning about "Variable variable usage", so I decided to make it explicit to avoid the warning. If you wish we can change it back.

Reply:
I think you mostly escaped the tool inspections, but it's still a variable variable, by design :D

Reply:
hehehe, touché! I reverted it.